examples/button.rs

```diff
@@ -0,0 +1,158 @@
+
+
+    window.set_ups(60);
+
+    // Poll events from the window.
+    while let Some(event) = window.next() {
+
+        // Convert the piston event to a conrod event.
+        if let Some(e) = conrod::backend::piston_window::convert_event(event.clone(), &window) {
+            ui.handle_event(e);
+        }
+
+        // We'll set all our widgets in a single function called `set_widgets`.
+        event.update(|_| {
+            ui.set_widgets(|mut ui| {
+                set_widgets(&mut ui, &mut app);
+            });
+        });
+
+        // Draw our Ui!
+        //
+        // The `draw_if_changed` method only re-draws the GUI if some `Widget`'s `Element`
+        // representation has changed. Normally, a `Widget`'s `Element` should only change
+        // if a Widget was interacted with in some way, however this is up to the `Widget`
+        // designer's discretion.
+        //
+        // If instead you need to re-draw your conrod GUI every frame, use `Ui::draw`.
+        window.draw_2d(&event, |c, g| {
+            if let Some(primitives) = ui.draw_if_changed() {
+                fn texture_from_image<T>(img: &T) -> &T { img };
+                conrod::backend::piston_window::draw(c, g, primitives,
+                                                     &mut text_texture_cache,
+                                                     &image_map,
+                                                     texture_from_image);
+            }
+        });
+    }
+}
+
+// In conrod, each widget must have its own unique identifier so that the `Ui` can keep track
+// of its state between updates.
+//
+// To make this easier, conrod provides the `widget_ids` macro, which generates a unique
+// const `widget::Id` for each identifier given in the list.
+//
+// The `with n` syntax reserves `n` number of `widget::Id`s for that identifier, rather than
+// just one.
+//
+// This is often useful when you need to use an identifier in some kind of loop.
+widget_ids! {
+    CANVAS,
+    BUTTON,
+    RUST_LOGO
+}
+
+/// Set all `Widget`s within the User Interface.
+///
+/// The first time this gets called, each `Widget`'s `State` will be initialised and cached within
+/// the `Ui` at their given indices. Every other time this get called, the `Widget`s will avoid any
+/// allocations by updating the pre-existing cached state. A new graphical `Element` is only
+/// retrieved from a `Widget` in the case that it's `State` has changed in some way.
+fn set_widgets(ui: &mut conrod::UiCell, app: &mut Context) {
+    use conrod::{color, widget, Colorable, Positionable, Sizeable, Texturable, Widget};
+
+    // We can use this `Canvas` as a parent Widget upon which we can place other widgets.
+    widget::Canvas::new()
+        .pad(30.0)
+        .color(app.bg_color)
+        .scroll_kids()
+        .set(CANVAS, ui);
+    // Button widget example button.
+    widget::Button::new()
+        .w_h(200.0, 200.0)
+        .middle_of(CANVAS)
+        .rgb(0.4, 0.75, 0.6)
```

Comment:
If the `Button` is using an `Image`, does this coloring method do anything? Hmmm, we should work out some way so that if the user has specified an image, they either:
- cannot call `Colorable` methods
- if they do call `Colorable` methods, this sets the `color` method on the `Image` used inside the `Button`.

I have some ideas, but would probably be easier for me to do a follow up PR and CC you in it rather than explain them here.